main.tf

```diff
@@ -65,8 +55,8 @@ resource "spacelift_policy" "trigger_dependency" {
 resource "spacelift_policy" "push" {
   type = "GIT_PUSH"
 
-  name = "Component Push Policy"
-  body = file("${path.module}/policies/push-stack.rego")
+  name = "Global Push Policy"
+  body = file("${path.module}/policies/push-global.rego")
```

Comment:
Thinking we should use the "catalog" pattern (e.g. `catalog/policies/`) here. And then enable use some kind of convention like:

- `trigger-something.rego` becomes type `TRIGGER`
- `git_push-blah.rego` becomes type GIT_PUSH`'

See datadog module, SCP module, and aws config modules..

Reply:
then we can have a variable which is:
```
variable "policies_enabled" {
  type = list
  default = [ "something.trigger", "blah.git_push" ]
}

Reply:
@osterman So this is the only outstanding issue/request at the moment. I've done a bit of testing on the rest of the changes and everything appears to be in good shape.

The latter part of this ask is complicated by the fact that these policies aren't applied at the same scope. The global trigger policy is ONLY applied to the top-level stack, while the git/dependency policies are applied to all child stacks.

I could hypothetically make `policies_enabled` a `map` (or `list(map)`?), and add the ability to add a scope for each policy entry, but does this extra functionality justify the additional complexity?

In any case, I'm guessing I could still use the catalog pattern (e.g. even if we explicitly set the scope in the module) -- the "default" policies just won't be a configurable option. Additionally, we could use `policies_enabled` for extra policies that get applied to each child stack (and/or the top-level stack, but I'm struggling to imagine the use case at the moment).

Yeahhhh, so we have a few options here and I'm curious about your thoughts on all of that :)

Reply:
Ok, so thanks for clarifying the "scope" of the problem. Let's hold off for now.
We'll implement it in a follow-up PR, and it doesn't have to be right away. Perhaps next iteration with the latest client.